### DISCUSSION TITLE
Cross Blueprint Field conditions

### DISCUSSION THREAD

**Discussion by fighter3005** [statamic/cms#10756] [Q&A] [answered]
Hi, 
does anyone know of a way to to make a condition dependent on a field that is outside of the current blueprint? 
For example, if I want to enable/disable (show/hide) a field for all collection items depending on a switch that is located in a global config (separate blueprint).

I only found "root." in the documentation, but I cannot leave the blueprint with that (my understanding). Is this scenario even possible without writing php code? I imagine there might be a way through an addon? Maybe somebody can shine a light on this scenario.

fyi I am using react/astro, if that is relevant.

Thank you for pointing me in the right direction! :)

**Comment by robdekort** [comment]
You need to create a custom condition for this. Which needs custom JS and PHP I'm afraid.

**Reply by robdekort** [reply, marked as answer]
In the `boot` method of your Service Provider you could do something like this:
```php
View::composer('statamic::layout', function ($view) {
    if (auth()->guest()) {
        return;
    }

    Statamic::provideToScript([
        'my_custom_config' => GlobalSet::findByHandle('my_global_handle')->inDefaultSite()->get('my_global_field'),
    ]);
});
```
Then in cp.js:
```js
Statamic.booting(() => {
    Statamic.$conditions.add('my_custom_condition', () => {
        if (StatamicConfig.my_custom_config) {
            return true;
        }
    });
});
```
And then in your fieldset yaml:
```yaml
if:
  label: 'custom my_custom_condition'
```
https://statamic.dev/conditional-fields#custom-logic

**Reply by fighter3005** [reply]
Thank you, works like a charm!
